### DISCUSSION TITLE
Is there a specific schedule for moving to the next dev cycle?

### DISCUSSION THREAD

**Discussion by zakkak** [oracle/graal#3509] [Q&A] [answered]
Hello all,

I see that GraalVM moved to 21.3-dev yesterday (22 June). 

Looking at previous changes I fail to see a pattern so I would like to ask whether there is a fixed date for "freezing" the next release branch based on some schedule. If not, how is the move to the next dev cycle planned?

| version | dev-cycle start | release date | dev-cycle duration |
| -- | -- | -- | -- |
| 21.3 | 2021/06/22 | 2021/10/19 | ~4 months |
| 21.2 | 2021/04/14 | 2021/07/20 | ~3 months |
| 21.1 | 2020/12/06 | 2021/04/20 | ~4.5 months |
| 21.0 | 2020/10/26 | 2021/01/19 | ~3 months |
| 20.3 | 2020/07/22 | 2020/11/17 | ~4 months |

Thank you

**Comment by gilles-duboscq** [comment, marked as answer]
Hi,

The release date is "the third Tuesday of January, April, July and October".
The "freeze" (i.e., creation of the release branch stemming from master) happens on Friday 32 days before that, except for the January release where we try to anticipate the freeze by 2 weeks to account for the fact that a lot of people take vacation at the end of December.

Merging the version bump in master after that should happen at the beginning of the week that follows the freeze. As you have seen it has sometimes been delayed a lot, we're trying to do that in a more timely fashion now.

This should give us the following calendar

| version | dev-cycle start | core freeze date* | general freeze date| release date |
| -- | -- | -- | -- | -- |
| 21.2 | 2021/03/22 | 2021/06/11 | 2021/06/18 | 2021/07/20 |
| 21.3 | 2021/06/21 | 2021/09/10 | 2021/09/17 | 2021/10/19 |
| 22.0 | 2021/09/20 | 2021/11/26 | 2021/12/03 | 2022/01/18 |
| 22.1 | 2021/12/06 | 2022/03/11 | 2022/03/18 | 2022/04/19 |
| 22.2 | 2022/03/21 | 2022/06/10 | 2022/06/17 | 2022/07/19 |
| 22.3 | 2022/06/20 | 2022/09/09 | 2022/09/16 | 2022/10/18 |
| 23.0 | 2022/09/19 | 2023/03/10 | 2023/03/17 | 2023/06/? |
| 23.1 | 2023/03/24 | 2023/08/11 | 2023/08/18 | 2023/09/19 |

\***core freeze date**: During the week leading up to the general freeze, core components (Compiler, Native Image, and Truffle) should not merge new features and are effectively frozen.

**Reply by zakkak** [reply]
Thank you @gilles-duboscq that helps a lot :)

**Reply by gilles-duboscq** [reply]
One thing that might be worth noting:
During the week leading up to the freeze, we have a policy that core components (Compiler, Native Image, Truffle) should not merge new features.

**Reply by gilles-duboscq** [reply]
I have updated the answer to include the "core freeze date" and added one more row for 22.3

**Reply by zakkak** [reply]
Hi @gilles-duboscq, could you please update your answer to also mention the new (OpenJDK-like) freeze period and release dates? Thanks

**Reply by zakkak** [reply]
@gilles-duboscq this is a kind reminder
